posydon/binary_evol/CE/step_CEE.py

```diff
@@ -814,4 +826,92 @@ def CEE_simple_alpha_prescription(
+                                                        # = Mdonor,envelope / Mcomp,envelope == Mejected_donor / Mejected_comp
+
+                    # Mejected_comp = Mejected_donor / WF
+                    # A*Mejected_donor^2 + B*Mejected_donor + C = 0
+
+                    A = alpha_CE / (WF*2.*separation_postCEE)
+                    B = ( ( - alpha_CE*m1_i * const.Msun/ (WF*2.*separation_postCEE)) - (alpha_CE*m2_i * const.Msun/ (2.*separation_postCEE)) \
+                        - (m1_i * const.Msun/ (lambda1_CE*radius1 * const.Rsun)) - (m2_i * const.Msun/ (WF*lambda2_CE*radius2 * const.Rsun)))
+                    C = alpha_CE * (m1_i * const.Msun * m2_i * const.Msun)/2. * (1./separation_postCEE - 1./separation_i)
+                    roots = np.roots([A,B,C])
+                    if min( np.real(roots[0]), np.real(roots[1]) ) < 0.:
+                        root = max( np.real(roots[0]), np.real(roots[1]) ) # if one root is negative, the other is considered the physical solution
+                    elif max( np.real(roots[0]), np.real(roots[1]) ) > m1_i  - mc1_i: # if one root is more positive than the Menvelope that we allow to ejected
+                                                                                     # then the other is considered the physical solution
+                        root = min( np.real(roots[0]), np.real(roots[1]) )
+
+                    Mejected_donor = root  / const.Msun # to go from cgs to solar masses again
+                    Mejected_comp = Mejected_donor / WF
+
+                    if verbose:
+                        print("The roots are: ", roots / const.Msun, "in Msun")
+                        print("so Mejecta_donor = ", Mejected_donor, "in Msun compared to its initial envelope =",  m1_i  - mc1_i)
+                        print("so Mejecta_comp = ", Mejected_comp, "in Msun compared to its initial envelope =",  m2_i  - mc2_i)
+
+                    if not ( (Mejected_donor <= m1_i  - mc1_i) and (Mejected_comp <= m2_i  - mc2_i) ):
+                        raise Exception("Mejected_donor in double CEE with mass_loss_during_CEE_merged is found more than Menvelope")
+                    if not (Mejected_donor >= 0.):
+                        raise Exception("The root of the equation in double CEE with mass_loss_during_CEE_merged is found negative")
+
+
+                    if (Mejected_donor > m1_i  - mc1_i) or (Mejected_comp > m2_i  - mc2_i):
+                        print("MANOS", Mejected_donor > m1_i  - mc1_i, Mejected_comp > m2_i  - mc2_i)
+                        Mejected_donor = (m1_i  - mc1_i) -0.01 # at least this value of envelope is left.
+                        Mejected_comp = (m2_i  - mc2_i) -0.01
+                        warnings.warn("Mejected of at least one star in double CEE is found to be more that the initial envelope. Reduced both to their initial_envelope - 0.01 Msun")
+
+                donor.mass = m1_i - Mejected_donor
+                donor.log_R = np.nan
```

Comment:
I was thinking, whether we could/should set here a radius estimate:
Usually, I'd expect it to be the Roche-lobe radius, which was assumed to do all this calculation.

Reply:
I get your point, but it is not really needed because these stars will merge and the log_R will change even further. For now we are agnostic with radii of merged products, we mostly focus on mass and surface/center abundances to  make a crude estimation about the merged product